tests/unit/test_rest_streaming.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import random
+from typing import List
+from unittest.mock import patch
+
+import proto
+import pytest
+import requests
+
+from google.api_core import rest_streaming
+
+
+class Composer(proto.Message):
+    given_name = proto.Field(proto.STRING, number=1)
+    family_name = proto.Field(proto.STRING, number=2)
+
+
+class Song(proto.Message):
+    composer = proto.Field(Composer, number=1)
+    title = proto.Field(proto.STRING, number=2)
+    lyrics = proto.Field(proto.STRING, number=3)
+    year = proto.Field(proto.INT32, number=4)
+
+
+class EchoResponse(proto.Message):
+    content = proto.Field(proto.STRING, number=1)
+
+
+class ResponseMock(requests.Response):
+    class _ResponseItr:
+        def __init__(
+            self,
+            _response_bytes: bytes,
+            decode_unicode: bool,
+            random_split=False,
+            seed=0,
+        ):
+            self._responses_bytes = _response_bytes
+            self._i = 0
+            self._decode_unicode = decode_unicode
+            self._random_split = random_split
+            random.seed(seed)
+
+        def __next__(self):
+            if self._i == len(self._responses_bytes):
+                raise StopIteration
+            if self._random_split:
+                n = random.randint(1, len(self._responses_bytes[self._i :]))
```

Comment:
I think a randomized test is a good idea, but I also think it's important to keep track of the random seed in case we start seeing sporadic test failures. Maybe something like:

```python
import logging
import time

SEED = time.time_ns()
logging.info(f"Starting rest streaming tests with random seed: {SEED}")
random.seed(SEED)
```

Reply:
Done. mypy complains:
```
nox > mypy google tests
tests/unit/test_rest_streaming.py:31: error: Module has no attribute "time_ns"
Found 1 error in 1 file (checked 82 source files)
```

Reply:
Done.